src/trix/models/composition.coffee

```diff
@@ -69,52 +70,8 @@ class Trix.Composition extends Trix.BasicObject
     @setSelection(endPosition)
     @notifyDelegateOfInsertionAtRange([startPosition, endPosition])
 
-  breakFormattedBlock: ->
-    position = @getPosition()
-    range = [position - 1, position]
-
-    document = @document
-    {index, offset} = document.locationFromPosition(position)
-    block = document.getBlockAtIndex(index)
-
-    if block.getBlockBreakPosition() is offset
-      document = document.removeTextAtRange(range)
-      range = [position, position]
-    else
-      if block.text.getStringAtRange([offset, offset + 1]) is "\n"
-        range = [position - 1, position + 1]
-      else if offset - 1 isnt 0
-        position += 1
-
-    newDocument = new Trix.Document [block.removeLastAttribute().copyWithoutText()]
-    @setDocument(document.insertDocumentAtRange(newDocument, range))
-    @setSelection(position)
-
   insertLineBreak: ->
-    [startPosition, endPosition] = @getSelectedRange()
-    startLocation = @document.locationFromPosition(startPosition)
-    endLocation = @document.locationFromPosition(endPosition)
-    block = @document.getBlockAtIndex(endLocation.index)
-
-    if block.hasAttributes()
-      if block.isListItem()
-        if block.isEmpty()
-          @decreaseListLevel()
-          @setSelection(startPosition)
-        else if startLocation.offset is 0
-          document = new Trix.Document [block.copyWithoutText()]
-          @insertDocument(document)
-        else
-          @insertBlockBreak()
-      else
-        if block.isEmpty()
-          @removeLastBlockAttribute()
-        else if block.text.getStringAtRange([endLocation.offset - 1, endLocation.offset]) is "\n"
-          @breakFormattedBlock()
-        else
-          @insertString("\n")
-    else
-      @insertString("\n")
+    Trix.LineBreakInsertion.perform(this)
```

Comment:
@sstephenson - yeah I agree with that, especially since it's not clear by the name (`LineBreakInsertion`) that it contains insertion methods like `breakFormattedBlock`.

Since we're on the topic, what are your thoughts on [this comment](https://github.com/basecamp/trix/pull/280#issuecomment-237338645)? It's more of a question about the `$xManager` pattern. Not sure when that naming convention should be used, or if it applies here.